docs/ansible.netcommon.net_ping_module.rst

```diff
@@ -18,8 +18,7 @@ Version added: 1.0.0
 Synopsis
 --------
 - Tests reachability using ping from network device to a remote destination.
-- For Windows targets, use the :ref:`ansible.windows.win_ping <ansible.windows.win_ping_module>` module instead.
-- For targets running Python, use the :ref:`ansible.builtin.ping <ansible.builtin.ping_module>` module instead.
```

Comment:
@ashwini-mhatre I'm not sure if I totally understand why these lines are being removed and replaced with `wait_for`. Do you mind giving a quick overview of this?

Reply:
So this is fix for https://github.com/ansible-collections/ansible.netcommon/issues/100

Reply:
The behavior of built-in ping and net-ping is different. For net-ping we can use wait_for to achieve same behavior.

Reply:
`net_ping` (or any of the network platform specific ping modules) is used to check connectivity from the target node to another node and not between controller and target node. From my understanding, `wait_for` is not equivalent to net_ping and instead is a substitute for achieving ICMP ping check like behaviour. Neither network ping modules, not builtin ping module actually uses ICMP pings. Please correct me if I misunderstood something.

Reply:
builtin ping is not using ICMP pings please check https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ping_module.html

Reply:
also please check documentation for https://docs.ansible.com/ansible/latest/collections/ansible/netcommon/net_ping_module.html#ansible-collections-ansible-netcommon-net-ping-module if you check return values you will get to know this uses ICMP ping

Reply:
> builtin ping is not using ICMP pings please check https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ping_module.html

Yes, that is what I meant too.



> also please check documentation for https://docs.ansible.com/ansible/latest/collections/ansible/netcommon/net_ping_module.html#ansible-collections-ansible-netcommon-net-ping-module if you check return values you will get to know this uses ICMP ping

It does use ICMP pings, but not between controller and target. Instead, it tests the reachability between target and another node.

Reply:
ack